pickle/Numeric.java

```diff
@@ -0,0 +1,118 @@
+    {
+        // store the given result value string
+        strValue = resValue.strValue;
+        // validate that the given resValue is a numeric, raise exception on failure
+        parseString(scanner, strValue);
+    }
+
+    /**
+     * Overridden toString function to print Numeric Object
+     *
+     * @return String representation of Numeric Class
+     *
+     */
+    @Override
+    public String toString() {
+        return "Numeric{" +
+                "dataType=" + dataType +
+                ", strValue='" + strValue + '\'' +
+                ", doubleValue=" + doubleValue +
+                ", intValue=" + intValue +
+                '}';
+    }
+
+    /**
+     * Determines if a given string represents a Numeric value.
+     * <p>
+     *     1. If a string is one or more digits, it is considered
+     *     an integer.
+     * <p>
+     *     2. If a string is one or more digits, a '.', and one or
+     *     more digits, it is considered a float.
+     *
+     * @param scanner   Scanner object
+     * @param str       String to parse
+     */
+    private void parseString(Scanner scanner, String str)
+    {
+        // string is one or more digits
+        if (str.matches("\\d+"))
+        {
+            dataType = SubClassif.INTEGER;
+            storeValue(scanner, str, dataType);
+        }
+        // string is one or more digits, a '.', and one or more digits
+        else if (str.matches("\\d+\\.?\\d+"))
+        {
+            dataType = SubClassif.FLOAT;
+            storeValue(scanner, str, dataType);
+        }
+        // string is not an int or float
+        else
+        {
+            throw new Error("Value is not an int or float");
+            // TODO: raise exception via scanner.error()
+        }
+    }
+
+    /**
+     * Stores the given string value depending on the data type.
+     *
+     * @param str       String representation of Int or Flaot
+     * @param dataType  Subclassif.INTEGER or Subclassif.FLOAT
+     */
+    private void storeValue(Scanner scanner, String str, SubClassif dataType)
+    {
+        if (dataType == SubClassif.INTEGER)
+        {
+            // Try to parse the string into an integer
+            try {
+                intValue = Integer.parseInt(str);
+            }
+            catch (Exception ex)
+            {
+                // this likely means the integer in the string
+                // is too large for the int data type
+                throw new Error("Integer value could not be parsed");
+                // TODO: raise exception via scanner.error()
+            }
+        }
+        else if (dataType == SubClassif.FLOAT)
+        {
+            // Try to parse the string into a float
+            try {
+                doubleValue = Double.parseDouble(str);
+            }
+            catch (Exception ex)
+            {
+                throw new Error("Float value could not be parsed");
```

Comment:
I would say that if you wanted to throw an Error to stay consistent with the rest of the program to use the PickleException class to throw a generic error message, you can also use the NumericConstantException since you have access to the scanner.token and scanner.sourceFileNm which will both be protected in the scanner push. This will format an output like:

" Line # Integer value could not be parsed: ' tokenStr '  File: sourceFile  "